From 3d74e73994ea2dd7a44208bb6854debcbf383bb6 Mon Sep 17 00:00:00 2001
From: dengbo <bo.deng@spacemit.com>
Date: Wed, 23 Oct 2024 13:45:43 +0800
Subject: [PATCH 1193/1448] Fix dma_buf warning with enabled lockdep

Fix:
WARNING: CPU: 4 PID: 1697 at drivers/dma-buf/dma-buf.c:1109 dma_buf_map_attachment+0x7e/0xcc
Modules linked in: algif_hash algif_skcipher af_alg 8852bs binfmt_misc cdc_ncm ax_usb_nic ip_tables autofs4
CPU: 4 PID: 1697 Comm: gnome-shell Tainted: G        W          6.6.36 #20241023
Hardware name: spacemit k1-x deb1 board (DT)
epc : dma_buf_map_attachment+0x7e/0xcc
 ra : dma_buf_map_attachment+0x7c/0xcc
epc : ffffffff808a7f4a ra : ffffffff808a7f48 sp : ffffffc80be0b900
 gp : ffffffff82341fe0 tp : ffffffd945880000 t0 : 0000000000000001
 t1 : 0000000000000001 t2 : 0000000000000020 s0 : ffffffc80be0b930
 s1 : ffffffd91902b9c0 a0 : 0000000000000000 a1 : ffffffd90419d1b0
 a2 : 0000000000000000 a3 : 0000000000000001 a4 : ffffffda755f2908
 a5 : ffffffff8122d908 a6 : 0000000005d05dd5 a7 : 0000000000000267
 s2 : ffffffff8084e1a8 s3 : 0000000000000000 s4 : 00000000007e9000
 s5 : 0000000000000001 s6 : ffffffc80be0bb34 s7 : ffffffd91b8b8020
 s8 : 0000000000000020 s9 : ffffffd91902b788 s10: 0000000000000001
 s11: ffffffd945a1b400 t3 : ffffffda755f2908 t4 : 0000000000000000
 t5 : 0000000000000068 t6 : ffffffff82287510
status: 0000000200000120 badaddr: 0000000000000000 cause: 0000000000000003
[<ffffffff808a7f4a>] dma_buf_map_attachment+0x7e/0xcc
[<ffffffff8084e620>] PhysmemCreateNewDmaBufBackedPMR+0xfc/0x402
[<ffffffff8084ecd2>] PhysmemImportSparseDmaBuf+0x23a/0x378
[<ffffffff8084ee7e>] PhysmemImportDmaBuf+0x6e/0xd8
[<ffffffff807ddeec>] PVRSRVBridgePhysmemImportDmaBuf+0xfa/0x29a
[<ffffffff80810b36>] BridgedDispatchKM+0x134/0x480
[<ffffffff80853a12>] PVRSRV_BridgeDispatchKM+0x9a/0x156
[<ffffffff8079d41a>] drm_ioctl_kernel+0xa0/0x124
[<ffffffff8079d66a>] drm_ioctl+0x1cc/0x406
[<ffffffff8026341e>] __riscv_sys_ioctl+0x86/0xae
[<ffffffff80faa8c8>] do_trap_ecall_u+0xba/0x12e
[<ffffffff80fb65d2>] ret_from_exception+0x0/0x6e
---[ end trace 0000000000000000 ]---

Change-Id: I2dc5ccd1c8e85367b40b5ffe58d8e93a6801ce18
---
 drivers/gpu/drm/img-rogue/physmem_dmabuf.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/gpu/drm/img-rogue/physmem_dmabuf.c b/drivers/gpu/drm/img-rogue/physmem_dmabuf.c
index 619b9b5b4e4d..d11e71f7b299 100644
--- a/drivers/gpu/drm/img-rogue/physmem_dmabuf.c
+++ b/drivers/gpu/drm/img-rogue/physmem_dmabuf.c
@@ -317,7 +317,7 @@ static void PMRFinalizeDmaBuf(PMR_IMPL_PRIVDATA pvPriv)
 
 	psPrivData->ui32PhysPageCount = 0;
 
-	dma_buf_unmap_attachment(psAttachment, psSgTable, DMA_BIDIRECTIONAL);
+	dma_buf_unmap_attachment_unlocked(psAttachment, psSgTable, DMA_BIDIRECTIONAL);
 
 	if (psPrivData->bPoisonOnFree)
 	{
@@ -645,7 +645,7 @@ PhysmemCreateNewDmaBufBackedPMR(PHYS_HEAP *psHeap,
 		}
 	}
 
-	table = dma_buf_map_attachment(psAttachment, DMA_BIDIRECTIONAL);
+	table = dma_buf_map_attachment_unlocked(psAttachment, DMA_BIDIRECTIONAL);
 	if (IS_ERR_OR_NULL(table))
 	{
 		eError = PVRSRV_ERROR_INVALID_PARAMS;
@@ -761,7 +761,7 @@ PhysmemCreateNewDmaBufBackedPMR(PHYS_HEAP *psHeap,
 	return PVRSRV_OK;
 
 errUnmap:
-	dma_buf_unmap_attachment(psAttachment, table, DMA_BIDIRECTIONAL);
+	dma_buf_unmap_attachment_unlocked(psAttachment, table, DMA_BIDIRECTIONAL);
 errFreePhysAddr:
 	OSFreeMem(psPrivData->pasDevPhysAddr);
 errFreePrivData:
-- 
2.47.0

